### DISCUSSION TITLE
Switch to Rails.application.executor

### DISCUSSION THREAD

**Discussion by reidmorrison** [rocketjob/rocketjob#35] [Ideas]
RocketJob currently uses `::ActiveRecord::Base.clear_active_connections!` to clear out database connections.
Rails has moved to `Rails.application.executor`, see https://guides.rubyonrails.org/threading_and_code_execution.html.

This should resolve the issue where a db connection is rendered unusable if for example a connection timeout or other connection failure occurs.

@sudheer-meka would you like to give it a try? Sounds like a promising solution to the db connection failure issues.

**Comment by sudheer-meka** [comment]
Sure Reid,

Let me try this.

Thank you 🙏

**Reply by reidmorrison** [reply]
Also found this database.yml config option that may work with C Ruby with Postgres:
~~~yaml
reaping_frequency: 10
~~~

**Comment by sudheer-meka** [comment]
Yea, we have already added this. Which is giving us better results.
